### PR TITLE
Drop support for Shoots with Kubernetes version == 1.24

### PR DESCRIPTION
**Pull request by Kostov6**
**How to categorize this PR?**

/kind cleanup

**What this PR does / why we need it**:
- Add `unhealthyPodEvictionPolicy: AlwaysAllow` to PDBs
- Drop support for Shoots, Seeds and Gardens with Kubernetes version == 1.24.

**Which issue(s) this PR fixes**:
Part of https://github.com/gardener/gardener/issues/8921
Part of https://github.com/gardener/gardener/issues/8821
Part of #2 

**Special notes for your reviewer**:

**Release note**:

```breaking operator
`extension-shoot-rsyslog-relp` no longer supports Shoots with Кubernetes version == 1.24.
```

### CODE REVIEW

**Comment by shafeeqes** [comment]
There are more charts requiring cleanup, see https://github.com/gardener/gardener-extension-shoot-rsyslog-relp/pull/55/commits/7037bbc31f6053025c829d5686d3cecd5f13d37d

File: pkg/component/rsyslogrelpconfigcleaner/rsyslog_relp_config_cleaner.go [hunk 1]
```diff
@@ -93,94 +88,9 @@ func (r *rsyslogRelpConfigCleaner) WaitCleanup(ctx context.Context) error {
 
 func (r *rsyslogRelpConfigCleaner) computeResourcesData() (map[string][]byte, error) {
 	var (
-		objects            []client.Object
-		serviceAccountName = "default"
+		objects []client.Object
```
Comment (plkokanov) [inline pkg/component/rsyslogrelpconfigcleaner/rsyslog_relp_config_cleaner.go:91]: You can remove this variable and just pass the `daemonSet` directly here: https://github.com/gardener/gardener-extension-shoot-rsyslog-relp/blob/6dc16d1721fbd2fb9ff260aaebf35e5b868811f8/pkg/component/rsyslogrelpconfigcleaner/rsyslog_relp_config_cleaner.go#L168

**Review by plkokanov — CHANGES_REQUESTED** [review]
Thanks for the PR!
Just one minor nit inline.

**Pull request merged**